src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -405,13 +405,27 @@ def _build_storage_profile():
 def _build_data_disks(profile, data_disk_sizes_gb, image_data_disks,
                       data_caching, storage_sku, attach_data_disks=None):
     lun = 0
+
+    # handle 2 kinds of values
+    # 1 "--data-disk-caching <value>": all disks will be applied
+    # 2 "--data-disk-caching 1=<value> 2=<value>": apply based on lun, the rest will use server side default
+    default_caching, individual_disk_cachings = None, {}
+    if data_caching:
+        if len(data_caching) == 1 and '=' not in data_caching[0]:
+            default_caching = data_caching[0]
+        else:
+            for x in data_caching:
+                temp, caching = x.split('=', 1)
+                temp = int(temp)
+                individual_disk_cachings[temp] = caching
```

Comment:
Why do we not validate the caching value against the enum?

Reply:
I thought we can just leave to service end to figure out, but I am fine to add a bit more client side validation to get all errors clear. I have pushed out a new commit